src/ray/gcs/gcs_server/test/gcs_resource_scheduler_test.cc

```diff
@@ -90,13 +100,66 @@ class GcsResourceSchedulerTest : public ::testing::Test {
     CheckClusterAvailableResources(node_id, cpu_resource, node_cpu_num);
   }
 
+  void TestBinPackingByPriority(const gcs::SchedulingType &scheduling_type) {
+    // Add node resources.
+    std::string cpu_resource = "CPU";
+    std::string gpu_resource = "GPU";
+    std::string mem_resource = "memory";
+
+    std::vector<std::vector<std::pair<std::string, double>>> resources_list;
+
+    resources_list.emplace_back(std::vector({std::make_pair(cpu_resource, 1.0)}));
+    resources_list.emplace_back(std::vector({std::make_pair(cpu_resource, 2.0)}));
+    resources_list.emplace_back(std::vector(
+        {std::make_pair(cpu_resource, 1.0), std::make_pair(gpu_resource, 1.0)}));
+    resources_list.emplace_back(std::vector(
+        {std::make_pair(cpu_resource, 1.0), std::make_pair(gpu_resource, 2.0)}));
+    resources_list.emplace_back(std::vector(
+        {std::make_pair(cpu_resource, 1.0), std::make_pair(mem_resource, 1.0)}));
+    resources_list.emplace_back(std::vector(
+        {std::make_pair(cpu_resource, 1.0), std::make_pair(mem_resource, 2.0)}));
+
+    std::vector<NodeID> node_ids;
+    for (int i = 0; i < 6; i++) {
+      node_ids.emplace_back(NodeID::FromRandom());
+      AddClusterResources(node_ids.back(), resources_list[i]);
```

Comment:
Can you actually insert from the backward? 5->0. I think that will be close to generating the error scenario? Also did this test fail before this PR?

Reply:
Test failed before PR (you can comment out sort).